source/request-as-event-emitter.js

```diff
@@ -91,13 +91,11 @@ module.exports = (options = {}) => {
 				return;
 			}
 
-			setImmediate(() => {
-				try {
-					getResponse(response, options, emitter, redirects);
-				} catch (error) {
-					emitter.emit('error', error);
-				}
-			});
+			try {
+				getResponse(response, options, emitter, redirects);
+			} catch (error) {
+				emitter.emit('error', error);
+			}
```

Comment:
Can we leave this unchanged as this is being discussed in #525? Try not to duplicate PRs.

Reply:
I agree, this should not have been included here, but since I'm gonna accept #525, I'll allow it.